install_config/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,427 @@
+----
+====
+
+[[sssd-phase-2-authenticating-proxy-setup]]
+== Phase 2: Authenticating Proxy Setup
+
+This section guides you through the steps to authenticate the proxy setup.
+
+[[phase-2-step-1-copy-certificates]]
+=== Step 1: Copy Certificates
+
+From *_openshift.example.com_*, securely copy the necessary certificates to the
+proxy machine:
+
+====
+----
+# scp /etc/origin/proxy/master/ca.crt \
+      root@proxy.example.com:/etc/pki/CA/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.crt \
+      /etc/origin/proxy/authproxy.pem \
+      root@proxy.example.com:/etc/pki/tls/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.key \
+      root@proxy.example.com:/etc/pki/tls/private/
+----
+====
+
+[[phase-2-step-2-sssd-configuration]]
+=== Step 2: SSSD Configuration
+Install a new VM with a recent operating system. Run SSSD 1.12.0 or later so
+that you can use the *mod_identity_lookup* module. The examples in this topic
+use a Red Hat Enterprise Linux 7.2 Server.
+
+Install all of the necessary dependencies:
+
+====
+----
+# yum install -y sssd \
+                 sssd-dbus \
+                 realmd \
+                 httpd \
+                 mod_session \
+                 mod_ssl \
+                 mod_lookup_identity \
+                 mod_authnz_pam
+----
+====
+
+This gives you the needed SSSD and the web server components. Now, set up SSSD
+to authenticate this VM against the LDAP server. If the LDAP server is a FreeIPA
+or Active Directory environment, then *realmd* can be used to join this machine
+to the domain.
+
+====
+----
+# realm join ldap.example.com
+----
+====
+
+If you are not running a domain, then use the *authconfig* tool:
+
+====
+----
+# authconfig --update --enablesssd --enablesssdauth \
+             --ldapserver=ldap.example.com \
+             --enableldaptls \
+             --ldaploadcert=http://ldap.example.com/ca.crt
+----
+====
+
+This will create *_/etc/sssd/sssd.conf_* with most of the appropriate settings.
+
+[NOTE]
+====
+In RHEL 7, *authconfig* does not create the *_/etc/openldap/cacerts_* directory,
+so you may need to create it manually before running the above command.
+====
+
+If you want to use SSSD to manage failover situations for LDAP, this can be
+configured simply by adding additional entries in *_/etc/sssd/sssd.conf_* on the
+*ldap_uri* line. Systems enrolled with FreeIPA will automatically handle
+failover using DNS SRV records.
+
+By default, SSSD only collects a limited set of attributes about the user from
+the LDAP server (those necessary for local login information). The
+authenticating proxy will want to add the users’ email addresses to its
+output, so you need to configure SSSD to retrieve those as well.
+
+Add the following to the end of the *[domain/domainname]* section of
+*_/etc/sssd/sssd.conf_*:
+
+====
+----
+ldap_user_extra_attrs = mail
+----
+====
+
+Create the *[ifp]* section if it is not present:
+
+====
+----
+[ifp]
+user_attributes = +mail
```

Comment:
This came from an early draft. Please see https://sgallagh.wordpress.com/2016/05/27/openshift-and-sssd-part-3-extended-ldap-attributes/ for updated contents of these sssd.conf sections.